src/core/document.js

```diff
@@ -33,8 +32,8 @@ var Page = (function PageClosure() {
   var LETTER_SIZE_MEDIABOX = [0, 0, 612, 792];
 
   function isAnnotationRenderable(annotation, intent) {
-    return (intent === 'display' && annotation.viewable) ||
-           (intent === 'print' && annotation.printable);
+    return (intent === 'display' && annotation && annotation.viewable) ||
```

Comment:
This should not be possible, because `annotation` should always be defined. The annotations in the list returned by the annotations API must always contain valid `Annotation` objects, so what is the reason for this change?

Reply:
Here is a deal. There is a test file called annotation-text-without-popup.pdf that failed without this validation. Let me take a look once again and try to find why empty annotation was returned for this form

Reply:
So I took a look at this PDF in RUSP and as you can see it contains null annotation. That's why I added such a condition. Formally this is malformed document but I'd prefer to improve the annotation parsing a little bit

![image](https://user-images.githubusercontent.com/1661986/35753122-705a38c0-082c-11e8-8d74-d0884d752d3b.png)